crypto/fipsmodule/service_indicator/service_indicator_test.cc

```diff
@@ -1117,6 +1118,512 @@ TEST_P(HMACServiceIndicatorTest, HMACTest) {
+    // C, thus the hard-coded sizes in this section.
+    /* TODO(CryptoAlg-1281): not sure if these are actually Approved
+    {
+        EVP_sha1,
+        kHKDF_ikm_tc3, sizeof(kHKDF_ikm_tc3),
+        kHKDF_salt_tc3, 0,
+        kHKDF_info_tc3, 0,
+        kHKDF_okm_tc3_sha1, sizeof(kHKDF_okm_tc3_sha1),
+        AWSLC_APPROVED,
+    },
+    {
+        EVP_sha224,
+        kHKDF_ikm_tc3, sizeof(kHKDF_ikm_tc3),
+        kHKDF_salt_tc3, 0,
+        kHKDF_info_tc3, 0,
+        kHKDF_okm_tc3_sha224, sizeof(kHKDF_okm_tc3_sha224),
+        AWSLC_APPROVED,
+    },
+    {
+        EVP_sha256,
+        kHKDF_ikm_tc3, sizeof(kHKDF_ikm_tc3),
+        kHKDF_salt_tc3, 0,
+        kHKDF_info_tc3, 0,
+        kHKDF_okm_tc3_sha256, sizeof(kHKDF_okm_tc3_sha256),
+        AWSLC_APPROVED,
+    },
+    {
+        EVP_sha384,
+        kHKDF_ikm_tc3, sizeof(kHKDF_ikm_tc3),
+        kHKDF_salt_tc3, 0,
+        kHKDF_info_tc3, 0,
+        kHKDF_okm_tc3_sha384, sizeof(kHKDF_okm_tc3_sha384),
+        AWSLC_APPROVED,
+    },
+    {
+        EVP_sha512,
+        kHKDF_ikm_tc3, sizeof(kHKDF_ikm_tc3),
+        kHKDF_salt_tc3, 0,
+        kHKDF_info_tc3, 0,
+        kHKDF_okm_tc3_sha512, sizeof(kHKDF_okm_tc3_sha512),
+        AWSLC_APPROVED,
+    },
+     */
+
+    // RFC 5869 Test Case 4
+    {
+        EVP_sha1,
+        kHKDF_ikm_tc4, sizeof(kHKDF_ikm_tc4),
+        kHKDF_salt_tc4, sizeof(kHKDF_salt_tc4),
+        kHKDF_info_tc4, sizeof(kHKDF_info_tc4),
+        kHKDF_okm_tc4_sha1, sizeof(kHKDF_okm_tc4_sha1),
+        AWSLC_APPROVED,
+    },
+    {
+        EVP_sha224,
+        kHKDF_ikm_tc4, sizeof(kHKDF_ikm_tc4),
+        kHKDF_salt_tc4, sizeof(kHKDF_salt_tc4),
+        kHKDF_info_tc4, sizeof(kHKDF_info_tc4),
+        kHKDF_okm_tc4_sha224, sizeof(kHKDF_okm_tc4_sha224),
+        AWSLC_APPROVED,
+    },
+    {
+        EVP_sha256,
+        kHKDF_ikm_tc4, sizeof(kHKDF_ikm_tc4),
+        kHKDF_salt_tc4, sizeof(kHKDF_salt_tc4),
+        kHKDF_info_tc4, sizeof(kHKDF_info_tc4),
+        kHKDF_okm_tc4_sha256, sizeof(kHKDF_okm_tc4_sha256),
+        AWSLC_APPROVED,
+    },
+    {
+        EVP_sha384,
+        kHKDF_ikm_tc4, sizeof(kHKDF_ikm_tc4),
+        kHKDF_salt_tc4, sizeof(kHKDF_salt_tc4),
+        kHKDF_info_tc4, sizeof(kHKDF_info_tc4),
+        kHKDF_okm_tc4_sha384, sizeof(kHKDF_okm_tc4_sha384),
+        AWSLC_APPROVED,
+    },
+    {
+        EVP_sha512,
+        kHKDF_ikm_tc4, sizeof(kHKDF_ikm_tc4),
+        kHKDF_salt_tc4, sizeof(kHKDF_salt_tc4),
+        kHKDF_info_tc4, sizeof(kHKDF_info_tc4),
+        kHKDF_okm_tc4_sha512, sizeof(kHKDF_okm_tc4_sha512),
+        AWSLC_APPROVED,
+    },
+
+    // RFC Test Case 5 repeats the inputs from RFC 5869 Test Case 2; RFC Test
+    // Case 6 repeats the inputs from RFC 5869 Test Case 3.
+
+    // RFC 5869 Test Case 7
+    //
+    // Info is a zero-length array, thus the hard-coded length.
+    {
+        EVP_sha1,
+        kHKDF_ikm_tc7, sizeof(kHKDF_ikm_tc7),
+        NULL, 0,
```

Comment:
Should we not comment out the NULL salt for now with a TODO pointing to CryptoAlg-1281?

Reply:
I'm going to not touch this anymore until we hear back from atsec; I'll convert this PR into a draft so people can ignore it until I've incorporated atsec's suggestions.